test/utils/Partition_One_Test.cpp

```diff
@@ -0,0 +1,162 @@
+
+    ~PartitionOneTest() override {}
+
+    std::string file_search(const std::vector<std::string> &parent_dir_options, const std::string& file_basename)
+    {
+        // Build vector of names by building combinations of the path and basename options
+        std::vector<std::string> name_combinations;
+
+        // Build so that all path names are tried for given basename before trying a different basename option
+        for (auto & path_option : parent_dir_options)
+            name_combinations.push_back(path_option + file_basename);
+
+        return utils::FileChecker::find_first_readable(name_combinations);
+    }
+
+    void read_file_generate_partition_data()
+    {
+        const std::string file_path = file_search(hydro_fabric_paths, "catchment_data.geojson");
+        if (boost::algorithm::ends_with(catchmentDataFile, "gpkg")) {
+          #ifdef NGEN_WITH_SQLITE3
+          catchment_collection = ngen::geopackage::read(catchmentDataFile, "divides", catchment_subset_ids);
+          #else
+          throw std::runtime_error("SQLite3 support required to read GeoPackage files.");
+          #endif
+        } else {
+          catchment_collection = geojson::read(file_path, catchment_subset_ids);
+        }
+
+        for(auto& feature: *catchment_collection)
+        {
+            std::string cat_id = feature->get_id();
+            partition_data.catchment_ids.emplace(cat_id);
+            std::string nex_id = feature->get_property("toid").as_string();
+            partition_data.nexus_ids.emplace(nex_id);
+        }
+    }
+
+    void SetUp() override;
+
+    void TearDown() override;
+
+    void setupArbitraryExampleCase();
+
+};
+
+void PartitionOneTest::SetUp() {
+    setupArbitraryExampleCase();
+}
+
+void PartitionOneTest::TearDown() {
+
+}
+
+void PartitionOneTest::setupArbitraryExampleCase() {
+    hydro_fabric_paths = {
+        "data/",
+        "./data/",
+        "../data/",
+        "../../data/",
+
+    };
+}
+
+TEST_F(PartitionOneTest, TestPartitionData_1a)
+{
+    read_file_generate_partition_data();
+
+    partition_one.generate_partition(catchment_collection);
+    PartitionData data_struct = partition_one.partition_data;
+    catchment_ids = data_struct.catchment_ids;
+
+    //check catchment partition
+    std::vector<std::string> cat_id_vec;
+    // convert unordered_set to vector
+    for (const auto& id: catchment_ids) {
+        cat_id_vec.push_back(id);
+    }
+
+    //sort ids
+    std::sort(cat_id_vec.begin(), cat_id_vec.end());
+    //create set of unique ids
+    std::set<std::string> unique(cat_id_vec.begin(), cat_id_vec.end());
+    std::set<std::string> duplicates;
+    //use set difference to identify all duplicates
+    std::set_difference(cat_id_vec.begin(), cat_id_vec.end(), unique.begin(), unique.end(), std::inserter(duplicates, duplicates.end()));
+    if( duplicates.size() > 0 ){
+        for( auto& id: duplicates){
+        }
+        exit(1);
+    }
+
+    ASSERT_TRUE(true);
```

Comment:
```suggestion
    for( auto& id: duplicates){
        // Add printing code as needed
    }

    ASSERT_EQ(duplicates.size(), 0);
```
Unit tests should never be calling `exit()` directly, since that kills the whole test process, and doesn't properly report the failed test result.

The loop over duplicates doesn't need to be conditional, since if there are none, it does nothing.

Same applies to 1b below.